server/remote_cache/content_addressable_storage_server/content_addressable_storage_server_test.go

```diff
@@ -90,3 +99,95 @@ func TestMalevolentCache(t *testing.T) {
 	assert.Equal(t, d.GetHash(), get.GetResponses()[0].GetDigest().GetHash())
 	assert.Equal(t, int32(gcodes.NotFound), get.GetResponses()[0].GetStatus().GetCode())
 }
+
+func TestTreeTruncation(t *testing.T) {
+	instanceName := ""
+	ctx := context.Background()
+	te := testenv.GetTestEnv(t)
+	ctx, err := prefix.AttachUserPrefixToContext(ctx, te)
+	if err != nil {
+		t.Errorf("error attaching user prefix: %v", err)
+	}
+
+	clientConn := runCASServer(ctx, te, t)
+	bsClient := bspb.NewByteStreamClient(clientConn)
+	casClient := repb.NewContentAddressableStorageClient(clientConn)
+
+	// Upload a child file.
```

Comment:
Might be a bit easier to grok w/ this summarizing comment before all the uploads:

```go
// Build up the following dir layout:
// / (root dir)
//   /dir1
//     /file1
//   /dir2 (should be skipped)
//     /file2
```

(Also nit: `s/child/file/` for the FileNode names)